intermediate_source/torchvision_tutorial.rst

```diff
@@ -0,0 +1,505 @@
+           # ensure that they are aligned
+           self.imgs = list(sorted(os.listdir(os.path.join(root, "PNGImages"))))
+           self.masks = list(sorted(os.listdir(os.path.join(root, "PedMasks"))))
+
+       def __getitem__(self, idx):
+           # load images ad masks
+           img_path = os.path.join(self.root, "PNGImages", self.imgs[idx])
+           mask_path = os.path.join(self.root, "PedMasks", self.masks[idx])
+           img = Image.open(img_path).convert("RGB")
+           # note that we haven't converted the mask to RGB,
+           # because each color corresponds to a different instance
+           # with 0 being background
+           mask = Image.open(mask_path)
+           # convert the PIL Image into a numpy array
+           mask = np.array(mask)
+           # instances are encoded as different colors
+           obj_ids = np.unique(mask)
+           # first id is the background, so remove it
+           obj_ids = obj_ids[1:]
+
+           # split the color-encoded mask into a set
+           # of binary masks
+           masks = mask == obj_ids[:, None, None]
+
+           # get bounding box coordinates for each mask
+           num_objs = len(obj_ids)
+           boxes = []
+           for i in range(num_objs):
+               pos = np.where(masks[i])
+               xmin = np.min(pos[1])
+               xmax = np.max(pos[1])
+               ymin = np.min(pos[0])
+               ymax = np.max(pos[0])
+               boxes.append([xmin, ymin, xmax, ymax])
+               
+           # convert everything into a torch.Tensor
+           boxes = torch.as_tensor(boxes, dtype=torch.float32)
+           # there is only one class
+           labels = torch.ones((num_objs,), dtype=torch.int64)
+           masks = torch.as_tensor(masks, dtype=torch.uint8)
+
+           image_id = torch.tensor([idx])
+           area = (boxes[:, 3] - boxes[:, 1]) * (boxes[:, 2] - boxes[:, 0])
+           # suppose all instances are not crowd
+           iscrowd = torch.zeros((num_objs,), dtype=torch.int64)
+
+           target = {}
+           target["boxes"] = boxes
+           target["labels"] = labels
+           target["masks"] = masks
+           target["image_id"] = image_id
+           target["area"] = area
+           target["iscrowd"] = iscrowd
+
+           if self.transforms is not None:
+               img, target = self.transforms(img, target)
+
+           return img, target
+
+       def __len__(self):
+           return len(self.imgs)
+
+That’s all for the dataset. Now let’s define a model that can perform
+predictions on this dataset.
+
+Defining your model
+-------------------
+
+In this tutorial, we will be using `Mask
+R-CNN <https://arxiv.org/abs/1703.06870>`__, which is based on top of
+`Faster R-CNN <https://arxiv.org/abs/1506.01497>`__. Faster R-CNN is a
+model that predicts both bounding boxes and class scores for potential
+objects in the image. 
+
+.. image:: ../../_static/img/tv_tutorial/tv_image03.png
```

Comment:
There are two placeholders in the colab notebook, the first one correspond to this image